code/modules/clothing/spacesuits/hardsuit.dm

```diff
@@ -507,3 +507,89 @@
                     "<span class='italics'>You hear loud electrical crackles.</span>")
     else
         return 0
+
+
+
+/////////////SHIELDED//////////////////////////////////
+
+/obj/item/clothing/suit/space/hardsuit/shielded
+	name = "shielded hardsuit"
+	desc = "A hardsuit with built in energy shielding. Will rapidly recharge when not under fire."
+	icon_state = "hardsuit-hos"
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/security/hos
+	allowed = list(/obj/item/device/flashlight,/obj/item/weapon/tank/internals, /obj/item/weapon/gun,/obj/item/weapon/reagent_containers/spray/pepper,/obj/item/ammo_box,/obj/item/ammo_casing,/obj/item/weapon/melee/baton,/obj/item/weapon/restraints/handcuffs)
+	armor = list(melee = 30, bullet = 15, laser = 30, energy = 10, bomb = 10, bio = 100, rad = 50)
+	var/current_charges = 3
+	var/max_charges = 1 //How many charges total the shielding has
+	var/recharge_delay = 50 //How long after we've been shot before we can start recharging. 20 seconds here
+	var/recharge_cooldown = 0 //Time since we've last been shot
+	var/recharge_rate = 1 //How quickly the shield recharges once it starts charging
+	var/shield_state = "shield-old"
+	var/shield_on = "shield-old"
+	var/datum/effect_system/spark_spread/sparks = new
+
+/obj/item/clothing/suit/space/hardsuit/shielded/New()
+	jetpack = new /obj/item/weapon/tank/jetpack/suit(src)
+	sparks.set_up(2, 1, src)
+	sparks.attach(src)
+	SSobj.processing |= src
+	..()
+
+/obj/item/clothing/suit/space/hardsuit/shielded/hit_reaction(mob/living/carbon/human/owner, attack_text)
+	if(current_charges > 0)
+		sparks.start()
+		owner.visible_message("<span class='danger'>[owner]'s shields deflect [attack_text] in a shower of sparks!</span>")
+		current_charges--
+		recharge_cooldown = world.time + recharge_delay
+		if(current_charges <= 0)
```

Comment:
i don't see anything that stops/breaks the shield completely,won't it just recharge normally even after breaking